framework/src/base/MooseApp.C

```diff
@@ -626,11 +626,11 @@ MooseApp::hasRecoverFileBase()
 }
 
 void
-MooseApp::meshOnly(std::string mesh_file_name)
+MooseApp::executeMeshOnlyActions()
```

Comment:
Hmm, this works but what happens when we modify this list? In fact, I probably will with mortar stuff soon. Can we possibly maintain a static list in MooseApp that you retrieve, modify and send back, or possibly inject new Actions into (like you do if you are registering a new task)? That might be a little safer. This is a little bit brittle.

Reply:
That was actually something in my mind, but it is almost equal to what this new function does. I think framework can just change this function. Apps will copy this function and insert new entries. If the function in framework changes, the function in apps need to be changed as well. Maybe a better solution is to let action warehouse maintain a list of mesh-only tasks and run the tasks in the correct order. Then application will just need to tell if a new registered task should be run in mesh-only or not.

Reply:
> Apps will copy this function and insert new entries. 

Which is on the list of things you should never do in code. 👿 
It's brittle and will break.

Reply:
I agree. Even with a static list, the way how new entries are inserted could need to be updated as well. If we want the action warehouse way, I will hand this over to you guys. I can override `meshOnly` as a temporary solution and change it later when the new capability is ready.